File: src/net/eduard/essentials/command/SetHomeCommand.java
package net.eduard.essentials.command;

import java.util.Set;

import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import net.eduard.api.lib.modules.Mine;
import net.eduard.api.lib.manager.CommandManager;
import net.eduard.essentials.EduEssentials;

public class SetHomeCommand extends CommandManager {
	public SetHomeCommand() {
		super("sethome");
	}
	public String message = "§bVoce setou um Home $home";
	public String messageNoPermissionMoreHomes = "§cVoce não tem permissão para setar mais Homes!";
	public int maxHomes = 100;
	@Override
	public boolean onCommand(CommandSender sender, Command command,
			String label, String[] args) {
		if (Mine.onlyPlayer(sender)) {

			Player p = (Player) sender;
			String home = "home";
			if (args.length >= 1) {
				home = args[0];
			}
			// home.limit.5
			String path = "homes."+p.getUniqueId().toString() + "." + home;
			Set<String> size = EduEssentials.getInstance().getConfigs().getKeys(p.getUniqueId().toString());
			int amount = size.size();
			if (!EduEssentials.getInstance().getConfigs().contains(path)) {
				if (!Mine.hasPerm(p, getPermission(), 100, amount + 1)) {

					p.sendMessage(
							messageNoPermissionMoreHomes);
					EduEssentials.getInstance().getConfigs().remove(path);
					return true;
				}
			}
			EduEssentials.getInstance().getConfigs().set(path, p.getLocation());
			sender.sendMessage(message.replace("$home", home));

		}

		return true;
	}

}

File: src/main/java/net/eduard/essentials/command/SetHomeCommand.kt
package net.eduard.essentials.command

import net.eduard.api.lib.manager.CommandManager
import net.eduard.api.lib.modules.Mine
import net.eduard.essentials.EduEssentials
import org.bukkit.entity.Player

class SetHomeCommand : CommandManager("sethome", "definircasa") {
    var message = "§bVoce setou um Home %home"
    var messageNoPermissionMoreHomes = "§cVoce não tem permissão para setar mais Homes!"
    var maxHomes = 100

    init {
        usage = "/sethome <nome>"
    }

    override fun playerCommand(player: Player, args: Array<String>) {
        var home = "home"
        if (args.isNotEmpty()) {
            home = args[0]
        }
        val path = "homes." + player.name.toLowerCase() + "." + home.toLowerCase()
        val size = EduEssentials.getInstance().storage.getKeys("homes."+player.name.toLowerCase())
        val amount = size.size
        if (!EduEssentials.getInstance().storage.contains(path)) {
            if (!Mine.hasPerm(player, permission, maxHomes, amount + 1)) {
                player.sendMessage(messageNoPermissionMoreHomes)
                EduEssentials.getInstance().storage.remove(path)
                return
            }
        }
        EduEssentials.getInstance().storage[path] = player.location
        player.sendMessage(message.replace("%home", home))
    }

}